packages/roosterjs-content-model-plugins/lib/paste/pasteSourceValidations/getPasteSource.ts

```diff
@@ -29,7 +30,8 @@ export type KnownPasteSourceType =
     | 'googleSheets'
     | 'wacComponents'
     | 'default'
-    | 'singleImage';
+    | 'singleImage'
+    | 'excelNonNativeEvent';
```

Comment:
Does this exist in all cases when copy from Excel? If so, do we still need to keep 'excelDesktop'? Maybe we just need to replace 'excelDesktop' and always use the new one.

Reply:
Yes, the shadow workbook type exists when copying from Excel.

The main difference is that when pasting wIth Ctrl + V we need the current approach, (clipboard fragment does not have a table element, so we do an special case).
But when getting the clipboard programatically, the fragment does contain a table but we need to add borders to the table.

Another option is to merge both logic in excel handling, but if fragment contains a table we do not need to build the table from the rawHtml

Reply:
Ok, then let's keep the current one to keep it simple.

Reply:
Oops, just modified the logic. Now we just keep all the excel handling in a single block of the switch. I think it looks cleaner. Let me know if it is okay